source/cppexpose/source/base/Tokenizer.cpp

```diff
@@ -0,0 +1,792 @@
+{
+    std::string buffer(token.begin, token.end-token.begin);
+    return helper::fromString<double>(buffer);
+}
+
+std::string Tokenizer::decodeString(const Token & token) const
+{
+    std::string decoded;
+    decoded.reserve(token.end - token.begin - 2);
+
+    const char * current = token.begin + 1; // Do not include '"'
+    const char * end     = token.end   - 1; // Do not include '"'
+
+    while (current != end)
+    {
+        char c = *current++;
+
+        if (c == '"')
+        {
+            break;
+        }
+
+        if (c == '\\')
+        {
+            if (current == end)
+            {
+                warning() << "Empty escape sequence in string";
+                return decoded;
+            }
+
+            char escape = *current++;
+            switch (escape)
+            {
+                case '"':  decoded += '"';  break;
+                case '/':  decoded += '/';  break;
+                case '\\': decoded += '\\'; break;
+                case 'b':  decoded += '\b'; break;
+                case 'f':  decoded += '\f'; break;
+                case 'n':  decoded += '\n'; break;
+                case 'r':  decoded += '\r'; break;
+                case 't':  decoded += '\t'; break;
+                case 'u':
+                {
+                    unsigned int unicode;
+
+                    if (!decodeUnicodeCodePoint(current, end, unicode)) {
+                        return "";
+                    }
+
+                    decoded += codePointToUTF8(unicode);
+                    break;
+                }
+
+                default:
+                {
+                    critical() << "Bad escape sequence in string";
+                    return "";
+                }
+            }
+        }
+        else
+        {
+            decoded += c;
+        }
+    }
+
+    return decoded;
+}
+
+bool Tokenizer::decodeUnicodeCodePoint(const char * & current, const char * end, unsigned int & unicode) const
+{
+    if (!decodeUnicodeEscapeSequence(current, end, unicode))
+    {
+        return false;
+    }
+
+    if (unicode >= 0xD800 && unicode <= 0xDBFF)
+    {
+        // surrogate pairs
+        if (end - current < 6) {
+            critical() << "Additional six characters expected to parse unicode surrogate pair.";
+            return false;
+        }
+
+        unsigned int surrogatePair;
+
+        if (*(current++) == '\\' && *(current++)== 'u') {
+            if (decodeUnicodeEscapeSequence(current, end, surrogatePair)) {
+                unicode = 0x10000 + ((unicode & 0x3FF) << 10) + (surrogatePair & 0x3FF);
+            } else {
+                return false;
+            }
+        } else {
+            critical() << "Expecting another \\u token to begin the second half of a unicode surrogate pair";
+            return false;
+        }
+    }
+
+    return true;
+}
+
+bool Tokenizer::decodeUnicodeEscapeSequence(const char * & current, const char * end, unsigned int & unicode) const
```

Comment:
Something like this is already in cppassist/string, so probably merge functionality? (https://github.com/cginternals/cppassist/blob/master/source/cppassist/include/cppassist/string/conversion.h)